src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -73,18 +74,18 @@ class LogInWithShortLivedTokenPage extends Component {
 
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+
+        // We have two routes in our history, "/" and "transition". We want to dismiss the trasition route, remove it from the history
+        // or else a user could be routed back to `/transition` when they close the modal.
+        // Or in some cases, the home screen gets stuck in `/transition` and there is an endless spinner in the background
+        CustomActions.navigateBackToRootDrawer();
```

Comment:
Why did we switch from `dismissModal()` to the custom action?

Reply:
I'm happy to just use `dismissModal()` here, but I was wondering if there was just a method within dismissModal we can just use instead. The old comments makes it seem unsure why we are even using this method here  `"But there is no current modal you say? I know, it confuses me too."`

`normalizedShouldOpenDrawer` is an argument we never pass in, so it would always be false. So I thought why don't we just use 
`CustomActions.navigateBackToRootDrawer();`

Reply:
The `CustomActions` are not used anywhere besides `Navigation` and `dismissModal()` should be performing the same logic. For consistency, we should just stick to using `Navigation.dismissModal()`?

Reply:
Just want to point out that a side effect of dismissing the modal immediately is that we won't wait for the policy to be created. I'm not sure if this is a blocker, but is it what we want @roryabraham @ctkochan22?

Reply:
Oh, that actually might be a minor problem, because I think that could result in a situation where we hide the transition page modal, flash the homepage, then open the workspace modal. That could open us up to weirdness, especially if the user interacts with the homepage (say, to open the global create menu), and then we automatically take them to their new workspace.

So maybe we should undo the changes in this file (other than comment clarifications, if desired), and instead close the modal once the new policy is created [here](https://github.com/Expensify/App/blob/ea07b4c694ed6ea7f1ab46404e6ad98aa1d8fade/src/libs/actions/Policy.js#L205)

Reply:
I was just doing that! I'll have the updated change up soon